hw/dv/data/dsim/dsim.hjson

```diff
@@ -2,8 +2,8 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 {
-  build_cmd:  "{job_prefix} {DSIM_HOME}/dsim"
-  run_cmd:    "{job_prefix} {DSIM_HOME}/dsim"
+  build_cmd:  "{job_prefix} {DSIM_HOME}/bin/dsim"
+  run_cmd:    "{job_prefix} {DSIM_HOME}/bin/dsim"
```

Comment:
I'm getting `No such file or directory` errors when using this `build_cmd` and `run_cmd` (after setting DSIM_HOME appropriately), can these be changed to `.../dsim.sh` ?

Reply:
Hi @udinator, the path to the DSim executable might be different depending on how it was installed for you. Normally $DSIM_HOME points to the directory where DSim is installed, and the executable is located in the bin directory directly below. Metrics doesn't provide a script called `dsim.sh`, however if this is something that has been set up by your tools group and works for you, please feel free to make the change.

Reply:
Unfortunately DSim binary does not work out of a box. It requires its shared libraries to be in LD_LIBRARY_PATH. Modifying the shell is not a good idea. It can cause an issue using different dsim versions.
That's exactly the reason why we introduced the dsim.sh script to take care of the LD_LIBRARY_PATH setup and dsim binary invocation. I am open to discuss alternatives.

Reply:
Maybe we can add the dsim.sh script to the `opentitan/hw/dv/data/dsim` directory? We are ready to open-source that script, and it would simplify things on both sides as it automatically sets up the `LD_LIBRARY_PATH` env variable.
@aimeepsutton @slaval-google what do you think?

Reply:
I have no objection. @slaval-google if you agree can you share the contents of dsim.sh with me? Otherwise I'm unable to test these changes locally. Thank you!

Reply:
The solution should work not only for HW engineers at Google, but pretty much anyone who has a DSim license. 

Ideally, the tool assumes that as a part of the workstation 'prep' step, the paths where the binaries and shared libs exist are already added to `$PATH` and $LD_LIBRARY_PATH` respectively, so this can be `build_cmd: {job_prefix} dsim`. 

If that is not an option, then we can export stuff to the shell's env using the exports key:
```
  exports: [
    LD_LIBRARY_PATH: "<dsim-lib-path>:{LD_LIBRARY_PATH}"
    PATH: "<dsim-bin-path>:{PATH}
  ]
```
I have not tried this though :)

Reply:
DSim setup could be complicated due to many dependencies. 
Let's assume the setup script takes care of:
- adding 'dsim' tool into the PATH;
- set up LD_LIBRARY_PATH;
- set up DSIM_HOME and UVM_HOME.

The OpenTitan config would be:
  build_cmd:  "{job_prefix} dsim"
  run_cmd:    "{job_prefix} dsim"

Reply:
Slava's suggestion is my preferred solution. It's also analogous to the way other simulators are invoked in the other <simulator>.hjson files. Various DSim users will have their own ways of managing tool versions and configuration and I don't think it's necessary to address that problem here. 

If all parties agree, I'll make the change that Slava suggests and add it to this PR.

Reply:
I think I'd prefer Slava's solution as well.

Reply:
Implemented Slava's suggestion here: https://github.com/lowRISC/opentitan/pull/2485/commits/989ffb4342dbd52105d6b424bf6139859fde6869